tests/migrations/test_loader.py

```diff
@@ -48,6 +48,16 @@ def test_apply(self):
             set(),
         )
 
+    def test_has_table_cached(self):
+        """
+        The has_table() method caches a positive result and not continually
+        query for the existence of the migrations table.
+        """
+        recorder = MigrationRecorder(connection)
+        with self.assertNumQueries(1):
```

Comment:
The assertion fails on Snowflake because `DatabaseIntrospection.get_table_list()` executes two queries (`SHOW TABLES` and `SHOW VIEWS`). We could add a database feature for the expected number of queries or check the number of queries after the first `has_table()` call and ensure no more occur after the second. I think the second option would be fine.

Reply:
Thanks for the report, see #17022.